server/athenian/api/internal/miners/jira/issue.py

```diff
@@ -435,15 +434,87 @@ async def _fetch_jira_issues(
     else:
         on_raw_fetch_complete_task = _NoResult.stub()
     if issues.empty or not adjust_timestamps_using_prs:
-        for col in (
-            ISSUE_PRS_BEGAN,
-            ISSUE_PRS_RELEASED,
-            ISSUE_PRS_COUNT,
-            ISSUE_PR_IDS,
-            resolved_colname,
-        ):
-            issues[col] = [None] * len(issues)
-        return issues, await on_raw_fetch_complete_task, adjust_timestamps_using_prs
+        _fill_issues_with_empty_prs_info(issues)
+    else:
+        assert default_branches is not None
+        await _fill_issues_with_mapped_prs_info(
+            issues,
+            default_branches,
+            release_settings,
+            logical_settings,
+            account,
+            meta_ids,
+            jira_filter.account,
+            pdb,
+            mdb,
+        )
+    return issues, await on_raw_fetch_complete_task, adjust_timestamps_using_prs
+
+
+@sentry_span
+async def fetch_jira_issues_by_keys(
+    keys: Sequence[str],
+    jira_config: JIRAConfig,
+    default_branches: dict[str, str],
+    release_settings: ReleaseSettings,
+    logical_settings: LogicalRepositorySettings,
+    account: int,
+    meta_ids: tuple[int, ...],
+    mdb: Database,
+    pdb: Database,
+    cache: Optional[aiomcache.Client],
+) -> pd.DataFrame:
+    """Load JIRA issues based on their key. Result dataframe order is undetermined."""
+    columns = [
+        Issue.id,
+        Issue.created,
+        AthenianIssue.updated,
+        AthenianIssue.work_began,
+        AthenianIssue.resolved,
+        Status.category_name,
+        Issue.key,
+        Issue.title,
+        Issue.reporter_display_name,
+        Issue.assignee_display_name,
+        Issue.url,
+        Issue.priority_name,
+        Issue.status,
+        Issue.type_id,
+        Issue.project_id,
+        Issue.comments_count,
+        Issue.story_points,
```

Comment:
Since this is a low-level API, we should consider doing the same as in `_fetch_jira_issues`: have two column sets, the obligatory columns that we need to map to PRs, for example, and `extra_columns`.